core-bundle/contao/classes/DataContainer.php

```diff
@@ -914,6 +914,18 @@ protected function generateGlobalButtons(/* DataContainerGlobalOperationsBuilder
 		$legacyCallback = function (DataContainerOperation $config) {
 			trigger_deprecation('contao/core-bundle', '5.6', 'Using a button_callback without DataContainerOperation object is deprecated and will no longer work in Contao 6.');
 
+			if (!\is_array($config['button_callback'] ?? null) && !\is_callable($config['button_callback'] ?? null))
+			{
+				return;
+			}
+
+			if ($config['icon'])
+			{
+				$icon = Controller::addAssetsUrlTo($config['icon']);
```

Comment:
Currently `$attributes` will only get

```
 style="background-image: url(&apos;/wrench.svg&apos;);"
```

and thus the icon will not be working. With the following change:

```suggestion
				$icon = $config['icon'];

				// Add the theme path if only the file name is given
				if (!str_contains($icon, '/'))
				{
					$icon = Image::getPath($icon);
				}

				$icon = Controller::addAssetsUrlTo($icon);
				$config['attributes']->addStyle("background-image: url('$icon')");
```

it will be

```
 style="background-image: url(&apos;/system/themes/flexible/icons/wrench.svg&apos;);"
```

This will apply the same logic as before.

https://github.com/contao/contao/blob/d3e720d1501a9e2d9b405c2342c9cdf63cd8cf1c/core-bundle/contao/classes/DataContainer.php#L1070-L1076